specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx

```diff
@@ -173,48 +58,61 @@ export function IntegratedRecordSelector({
 
   const [rawIndex, setIndex] = useSearchParameter(urlParameter);
   const index = f.parseInt(rawIndex) ?? 0;
+
+  const isInteraction =
+    typeof relationship === 'object' &&
+    relationshipIsToMany(relationship) &&
+    typeof collection.related === 'object' &&
+    ['LoanPreparation', 'GiftPreparation', 'DisposalPreparation'].includes(
+      relationship.relatedTable.name
+    );
+
+  const [isDialogOpen, handleOpenDialog, handleCloseDialog] = useBooleanState();
+
   return (
     <ReadOnlyContext.Provider value={isReadOnly}>
-      {formType === 'formTable' ? (
-        <FormTableCollection
-          collection={collection}
-          dialog={dialog}
-          sortField={sortField}
-          viewName={viewName}
-          onAdd={(resources): void => {
-            collection.add(resources);
-            if (typeof handleAdd === 'function') handleAdd(resources);
-          }}
-          onClose={handleClose}
-          onDelete={
-            handleDelete === undefined
-              ? undefined
-              : (_resource, index): void => handleDelete(index, 'minusButton')
-          }
-        />
-      ) : (
-        <RecordSelectorFromCollection
-          collection={collection}
-          defaultIndex={isToOne ? 0 : index}
-          relationship={relationship}
-          onAdd={handleAdd}
-          onDelete={handleDelete}
-          onSlide={(index): void =>
-            typeof urlParameter === 'string'
-              ? setIndex(index.toString())
-              : undefined
+      <RecordSelectorFromCollection
+        collection={collection}
+        defaultIndex={isToOne ? 0 : index}
+        relationship={relationship}
+        isInteraction={isInteraction}
+        onAdd={(resource) => {
+          handleOpenDialog();
+          if (handleAdd !== undefined) {
+            handleAdd(resource);
           }
-          {...rest}
-        >
-          {({
-            dialogs,
-            slider,
-            resource,
-            onAdd: handleAdd,
-            onRemove: handleRemove,
-            isLoading,
-          }): JSX.Element => (
-            <>
+        }}
+        onDelete={handleDelete}
+        onSlide={(index): void =>
+          typeof urlParameter === 'string'
+            ? setIndex(index.toString())
+            : undefined
+        }
+        {...rest}
+      >
+        {({
+          dialogs,
+          slider,
+          resource,
+          onAdd: handleAdd,
+          onRemove: handleRemove,
+          isLoading,
+        }): JSX.Element => (
+          <>
+            {isInteraction &&
+            typeof collection.related === 'object' &&
+            isDialogOpen ? (
+              <InteractionDialog
+                actionTable={collection.related?.specifyTable}
```

Comment:
You are checking for `typeof collection.related === 'object'` 3 lines above, so no need for `?` here